### PR TITLE
Refactor restore secret handling to fix spec_overrides bug

### PR DESCRIPTION
**Pull request by rooftopcellist**
##### SUMMARY
Fix spec_overrides bug and align this operator's backup and restore roles with the other ansible-related operators.  

##### ADDITIONAL INFORMATION
Alternative approach here:
* https://github.com/ansible/galaxy-operator/pull/126

### CODE REVIEW

File: roles/backup/tasks/dump_generated_secret.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+---
+
+- name: Get secret name
+  set_fact:
+      _name: "{{ status[item] }}"
```
Comment (dsavineau) [inline roles/backup/tasks/dump_generated_secret.yml:5]:
```suggestion
  set_fact:
    _name: "{{ status[item] }}"
```

File: roles/backup/tasks/dump_generated_secret.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      - name: Set error message
+        set_fact:
+            error_msg: "{{ item }} status is not set on {{ deployment_type }} object yet"
+
+      - name: Handle error
+        import_tasks: error_handling.yml
+
+      - name: Fail early if secret name status is not set
+        fail:
+            msg: "{{ error_msg }}"
```
Comment (dsavineau) [inline roles/backup/tasks/dump_generated_secret.yml:18]:
```suggestion
  block:
    - name: Set error message
      set_fact:
        error_msg: "{{ item }} status is not set on {{ deployment_type }} object yet"

    - name: Handle error
      import_tasks: error_handling.yml

    - name: Fail early if secret name status is not set
      fail:
        msg: "{{ error_msg }}"
```

File: roles/backup/tasks/dump_generated_secret.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  when: _name is not defined or _name == ''
+
+- name: Get secret
+  k8s_info:
+      version: v1
+      kind: Secret
+      namespace: '{{ ansible_operator_meta.namespace }}'
+      name: "{{ _name }}"
```
Comment (dsavineau) [inline roles/backup/tasks/dump_generated_secret.yml:26]:
```suggestion
  k8s_info:
    version: v1
    kind: Secret
    namespace: '{{ ansible_operator_meta.namespace }}'
    name: "{{ _name }}"
```

File: roles/backup/tasks/dump_generated_secret.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  register: _secret
+  no_log: "{{ no_log }}"
+
+- name: Set secret data
+  set_fact:
+      _data: "{{ _secret['resources'][0]['data'] }}"
+      _type: "{{ _secret['resources'][0]['type'] }}"
```
Comment (dsavineau) [inline roles/backup/tasks/dump_generated_secret.yml:33]:
```suggestion
  set_fact:
    _data: "{{ _secret['resources'][0]['data'] }}"
    _type: "{{ _secret['resources'][0]['type'] }}"
```

File: roles/backup/tasks/dump_generated_secret.yml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  no_log: "{{ no_log }}"
+
+- name: Create and Add secret names and data to dictionary
+  set_fact:
+      secret_dict: "{{ secret_dict | default({}) | combine({ item: {'name': _name, 'data': _data, 'type': _type }}) }}"
```
Comment (dsavineau) [inline roles/backup/tasks/dump_generated_secret.yml:38]:
```suggestion
  set_fact:
    secret_dict: "{{ secret_dict | default({}) | combine({ item: {'name': _name, 'data': _data, 'type': _type }}) }}"
```

File: roles/backup/tasks/dump_secret.yml [hunk 6]
```diff
@@ -0,0 +1,28 @@
+---
+
+- name: Get Secret Name
+  set_fact:
+      _name: "{{ galaxy_spec.spec[item] | default('') }}"
```
Comment (dsavineau) [inline roles/backup/tasks/dump_secret.yml:5]:
```suggestion
  set_fact:
    _name: "{{ galaxy_spec.spec[item] | default('') }}"
```

File: roles/backup/tasks/dump_secret.yml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+      - name: Set secret key
+        set_fact:
+            _data: "{{ _secret['resources'][0]['data'] }}"
+            _type: "{{ _secret['resources'][0]['type'] }}"
+        no_log: "{{ no_log }}"
+
+      - name: Create and Add secret names and data to dictionary
+        set_fact:
+            secret_dict: "{{ secret_dict | default({}) | combine({item: { 'name': _name, 'data': _data, 'type': _type }}) }}"
+        no_log: "{{ no_log }}"
```
Comment (dsavineau) [inline roles/backup/tasks/dump_secret.yml:27]:
```suggestion
  block:
    - name: Get secret
      k8s_info:
        version: v1
        kind: Secret
        namespace: '{{ ansible_operator_meta.namespace }}'
        name: "{{ _name }}"
      register: _secret
      no_log: "{{ no_log }}"

    - name: Set secret key
      set_fact:
        _data: "{{ _secret['resources'][0]['data'] }}"
        _type: "{{ _secret['resources'][0]['type'] }}"
      no_log: "{{ no_log }}"

    - name: Create and Add secret names and data to dictionary
      set_fact:
        secret_dict: "{{ secret_dict | default({}) | combine({item: { 'name': _name, 'data': _data, 'type': _type }}) }}"
      no_log: "{{ no_log }}"
```

**Review by dsavineau — COMMENTED** [review]
just some minor indent issues in two files otherwise looks good to me

**Comment by rooftopcellist** [comment]
Thanks for the review @dsavineau !

**Pull request merged**